### PR TITLE
chore: test for implemented variables

### PR DESCRIPTION
**Pull request by jptosso**
NOOP implemented Collection, but many functions were retrieving them using `tx.Collection(INVALID)` and casting them to Map, causing panic.

```
panic: interface conversion: *collections.noop is not collection.Map: missing method Add
```

### CODE REVIEW

File: internal/collections/noop.go [hunk 1]
```diff
@@ -12,10 +14,34 @@ var Noop collection.Collection = &noop{}
 
 type noop struct{}
 
+func (c *noop) Get(key string) []string {
```
Comment (jcchavezs) [inline internal/collections/noop.go:17]:
```suggestion
func (c *noop) Get(_ string) []string {
```

File: internal/collections/noop.go [hunk 2]
```diff
@@ -12,10 +14,34 @@ var Noop collection.Collection = &noop{}
...
+	return []string{}
+}
+
+func (c *noop) FindRegex(key *regexp.Regexp) []types.MatchData {
```
Comment (jcchavezs) [inline internal/collections/noop.go:21]:
```suggestion
func (c *noop) FindRegex(_ *regexp.Regexp) []types.MatchData {
```

File: internal/collections/noop.go [hunk 3]
```diff
@@ -12,10 +14,34 @@ var Noop collection.Collection = &noop{}
...
+	return []types.MatchData{}
+}
+
+func (c *noop) FindString(key string) []types.MatchData {
```
Comment (jcchavezs) [inline internal/collections/noop.go:25]:
```suggestion
func (c *noop) FindString(_ string) []types.MatchData {
```

File: internal/collections/noop.go [hunk 4]
```diff
@@ -12,10 +14,34 @@ var Noop collection.Collection = &noop{}
...
+	return []types.MatchData{}
+}
+
 func (c *noop) FindAll() []types.MatchData {
-	return nil
+	return []types.MatchData{}
 }
 
+func (c *noop) Add(key string, value string) {
+	c.Set(key, []string{value})
```
Comment (jcchavezs) [inline internal/collections/noop.go:34]: Is this right? Set is a noop

**Comment by anuraaga** [comment]
Where is it cast to Map? No code should cast a collection without type check

**Pull request closed without merging**